Src/IronPython/Runtime/Bytes.cs

```diff
@@ -36,8 +36,16 @@ public class Bytes : IList<byte>, IReadOnlyList<byte>, IEquatable<Bytes>, ICodeF
         }
 
         public Bytes([BytesLike, NotNull]IBufferProtocol source) {
-            using IPythonBuffer buffer = source.GetBuffer();
-            _bytes = buffer.AsReadOnlySpan().ToArray();
+            using IPythonBuffer buffer = source.GetBuffer(BufferFlags.FullRO);
```

Comment:
No idea if it would be used elsewhere, but could be an extension method:
```C#
public static byte[] ToArray(this IPythonBuffer buffer) {
    if (buffer.IsCContiguous()) {
        return buffer.AsReadOnlySpan().ToArray();
    } else {
        var bytes = new byte[buffer.NumBytes()];
        int i = 0;
        foreach (byte b in buffer.EnumerateBytes()) {
            bytes[i++] = b;
        }
        return bytes;
    }
}
```

Reply:
Looks good, and maybe it will be useful during pickling.